src/TerminalFighter/TextArea/TextArea.cpp

```diff
@@ -0,0 +1,24 @@
+#include "TextArea.h"
+
+TextArea::TextArea(std::string t, SDL_Color c, int x, int y, int w, int h, std::string font_path, int font_size)
+    : text_str_(t)
+    , color_(c)
+    , x_pos_(x)
+    , y_pos_(y)
+    , width_(w)
+    , height_(h)
+    , font_(TTF_OpenFont(font_path.c_str(), font_size)) {
+}
+
+TextArea::~TextArea() {}
+
+void TextArea::render(SDL_Renderer& renderer_) {
+    SDL_Surface* surface = TTF_RenderText_Solid(font_, text_str_.c_str(), color_);
+    SDL_Texture* Message = SDL_CreateTextureFromSurface(&renderer_, surface);
```

Comment:
Memory leak in this function.

You create a Surface, and then pass it into the Create Texture function - which is good. But then you don't delete the surface here. This surface will never be freed :(. gg

`SDL_FreeSurface(surface);`